cluster/restore-from-backup.sh

```diff
@@ -0,0 +1,188 @@
+#!/bin/bash
+
+# Copyright 2016 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This script performs disaster recovery of etcd from the backup data.
+# Assumptions:
+# - version.txt file is in the current directy (if it isn't it will be
+#     defaulted to "2.2.1/etcd2"). Based on this file, the script will
+#     decide to which version we are restoring (procedures are different
+#     for etcd2 and etcd3).
+# - in case of etcd2 - *.snap and *.wal files are in current directory
+# - in case of etcd3 - *.db file is in the current directory
+# - the script is run as root
+#
+# FIXME: The script currently doesn't support restoring of event etcd.
+
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+# Version file contains information about current version in the format:
+# <etcd binary version>/<etcd api mode> (e.g. "3.0.12/etcd3").
+#
+# If the file doesn't exist we assume "2.2.1/etcd2" configuration is
+# the current one and create a file with such configuration.
+# The restore procedure is choosed based on this information.
+VERSION_FILE="version.txt"
+if [ ! -f "${VERSION_FILE}" ]; then
+  echo "2.2.1/etcd2" > "${VERSION_FILE}"
```

Comment:
We are running 2.2.1 since half-an-year plus. That said, in my opinion we can asusme that we won't be restoring to backup from more than half-an-yesr ago.
@roberthbailey @fabioy - WDYT?